src/main/java/seedu/address/model/person/CovidStatus.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.stream.Stream;
+
+/**
+ * Represents a Person's covid status in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidCovidStatus(String)}
+ */
+public class CovidStatus {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Covid status should be one of the following: " + getCovidStatusEnumAsString();
+
+    public enum CovidStatusTier {
```

Comment:
I think that it is a great use of enum list here for the different covid statuses.